src/scantailor/core/zones.py

```
@@ -0,0 +1,285 @@
+
+        # For smooth splines, use cubic interpolation
+        from scipy.interpolate import splev, splprep
+
+        try:
+            # Close the spline by appending start point
+            pts = np.vstack([self.points, self.points[0]])
+            tck, _ = splprep([pts[:, 0], pts[:, 1]], s=0, per=True)
+            u = np.linspace(0, 1, num_samples)
+            x, y = splev(u, tck)
+            return np.column_stack([x, y])
+        except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)